### ISSUE TITLE
Please provide a way to sink to a standard Matrix without Tables or Dataframes

### ISSUE THREAD

**Issue by sylvaticus** [JuliaData/CSV.jl#1030]
Unless I am wrong (but if so, _many_ are on the internet too) to sink to a standard `Matrix` one has to first `using` Tables.jl or DataFrames.jl.
I believe it would be useful if CSV could be able to sink directly to a standard `Matrix`, e.g.
`foo= CSV.read("foo.csv", [options],Matrix)`

I am aware of `readdlm`, but CSV has way more options.

**Comment by quinnj** [comment]
Hmmm, yeah, we could probably support this really easily by just wrapping/unwrapping the `Tables.matrix` call for you with a specific `CSV.read` dispatch for `Matrix`. I think that makes sense. We originally decided _not_ to treat `Matrix` as a table in the Tables.jl interface sense because they don't really have column names/follow the other table properties, but I think this is a specific use-case where it's convenient to get a matrix out directly.

**Comment by quinnj** [comment]
@nickrobinson251, yeah that was my idea. Yeah, on the error path, maybe we have a `CSV.read(source; ::Type{<:AbstractMatrix})` that says we only support `Matrix` and to control types, pass `types=Int` or something?
